File: app/src/main/java/fr/free/nrw/commons/upload/GpsCategoryModel.java
package fr.free.nrw.commons.upload;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class GpsCategoryModel {
    private Set<String> categorySet;

    @Inject
    public GpsCategoryModel() {
        clear();
    }

    public void clear() {
        categorySet = new HashSet<>();
    }

    public boolean getGpsCatExists() {
        return !categorySet.isEmpty();
    }

    public List<String> getCategoryList() {
        return new ArrayList<>(categorySet);
    }

    public void setCategoryList(List<String> categoryList) {
        clear();
        categorySet.addAll(categoryList != null ? categoryList : new ArrayList<>());
    }
}

File: app/src/main/java/fr/free/nrw/commons/upload/GpsCategoryModel.kt
package fr.free.nrw.commons.upload

import fr.free.nrw.commons.category.CategoryItem
import io.reactivex.subjects.BehaviorSubject
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GpsCategoryModel @Inject constructor() {
    val categoriesFromLocation = BehaviorSubject.createDefault(emptyList<CategoryItem>())

    fun clear() {
        categoriesFromLocation.onNext(emptyList())
    }

    fun setCategoriesFromLocation(categoryList: List<CategoryItem>) {
        categoriesFromLocation.onNext(categoryList)
    }
}
